Add shareProcessNamespace flag to podOptions

gerlowskija opened the pull request:
Boolean flag defaults unset, and is available in the CRD at `.spec.customSolrKubeOptions.podOptions.shareProcessNamespace`

gerlowskija commented on helm/solr/Chart.yaml:
```diff
@@ -42,15 +42,12 @@ annotations:
   artifacthub.io/changes: |
     - kind: added
-      description: Addition 1
+      description: SolrClouds now support namespace sharing among pod containers in a pod.
```
Every time I come back to this project after a lull, I have trouble deciding which chart should hold a given change-entry.

Hopefully I've picked correctly here - I picked the 'solr' chart because the newly-exposed option affects the Solr pods (and not the operator pod itself)

HoustonPutman replied:
Solr Operator, since that is the main chart (you can use the solr operator chart without using the solr chart, but not really the other way around). The Solr chart changeling is only used for changes relating to that helm chart.

gerlowskija replied:
OK, thanks for clarifying; fixed!

gerlowskija commented:
This PR should allow us to close #716

HoustonPutman commented on api/v1beta1/common_types.go:
```diff
@@ -137,6 +137,10 @@ type PodOptions struct {
 
+	// Should process namespace sharing be enabled on created pods
+	// +optional
+	ShareProcessNamespace *bool `json:"shareProcessNamespace,omitempty"`
```
Since the default is false, this probably doesn't need to be a pointer. pointer is only used when we want to differentiate the default value of the variable and the default when users leave the field empty.

gerlowskija replied:
Makes sense; done.

HoustonPutman reviewed (commented):
this option should be mentioned in the Readme, `values.yaml` & `_custom_option_helpers.tpl` of the solr chart, since all the custom options are explicitly listed/set there.

gerlowskija commented:
> this option should be mentioned in the Readme, values.yaml & _custom_option_helpers.tpl of the solr chart, since all the custom options are explicitly listed/set there.

Oh, ok - I didn't realize we exposed every podOption there, but I guess that makes sense.  We still only want a Chart.yaml change-entry in the solr-operator helm chart I assume, per the in-line discussion above?

(Reminder to self: helm changes still need tested)

gerlowskija commented:
Alright, I think I addressed all your feedback @HoustonPutman , and I've tested out my changes to the 'solr' chart since my previous comment.  So this should be good to merge; I'll aim to do so shortly unless there's further feedback.

HoustonPutman commented on helm/solr/templates/_custom_option_helpers.tpl:
```diff
@@ -36,6 +36,9 @@ resources:
 serviceAccountName: {{ include "solr.serviceAccountName.solr" . }}
 {{ end }}
+{{- if .Values.podOptions.shareProcessNamespace -}}
+shareProcessNamespace: {{ .Values.podOptions.shareProcessNamespace}}
```
```suggestion
shareProcessNamespace: {{ .Values.podOptions.shareProcessNamespace }}
```
